xbmc/video/VideoDatabase.cpp

```diff
@@ -2862,9 +2881,20 @@ void CVideoDatabase::ClearBookMarksOfFile(int idFile, CBookmark::EType type /*=
 
     std::string strSQL=PrepareSQL("delete from bookmark where idFile=%i and type=%i", idFile, (int)type);
     m_pDS->exec(strSQL.c_str());
-    if (type == CBookmark::EPISODE)
+    if (CBookmark::IsMediaBookmark(type))
     {
-      strSQL=PrepareSQL("update episode set c%02d=-1 where idFile=%i", VIDEODB_ID_EPISODE_BOOKMARK, idFile);
+      switch (type)
```

Comment:
I can't use it.. DeleteBookMarkForMedia uses Database media id to delete the bookmark so it needs videoinfo. ClearBookMarksOfFile uses idfile path instead and doesn't have any video information